packages/peregrine/lib/talons/Checkout/useForm.js

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from '@apollo/react-hooks';
+
+export const useForm = props => {
+    const { countriesQuery } = props;
+
+    const {
+        loading: isLoadingCountries,
+        error: countriesError,
+        data: countriesData
+    } = useQuery(countriesQuery);
+    const { countries } = countriesData || {};
+
+    return {
+        countries,
+        countriesError,
+        isLoadingCountries
```

Comment:
I imagine we will want to make these APIs resilient to change as much as possible. To "future proof" this API you could just do something like:

```js
{
  isLoading: !!isLoadingCountries
  hasError: !!countriesError
}
```

And if we add cases for loading or error in the future we just have to extend them within the talon and the API won't change.

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/1993/commits/8ff820c2f73c711c8210e3b53b7ddaa287cbdbde.